### ISSUE TITLE
[bug] sveltekit vercel adapter does not bundle paraglide, leading to unresolvable imports

### ISSUE THREAD

**Issue by samuelstroschein** [inlang/monorepo#1485]
### Problem

The SvelteKit Vercel adapter does not bundle paraglide imports. 

1. The imports in the compiled output reference the node modules which don't exist in the vercel edge runtime. Every other import is referencing build output ("../chunk.js")
2. The directory structure of the build output does not include `node_modules` which likely leads to this problem. The paraglide import can't be resolved. 

![CleanShot 2023-10-18 at 10 18 40@2x](https://github.com/inlang/monorepo/assets/35429197/c0d215dd-5f41-42d7-b0db-fa91fba4fdb9)

### Expected behavior

The bundler processes the paraglide imports instead of leaving them as is. 

```diff
-import { availableLanguageTags, languageTag, setLanguageTag } from "@inlang/paraglide-js/website";
-import * as m from "@inlang/paraglide-js/website/messages";
+import { a, b, c, messageas } from "../chunk-2ss.js";
```

### Reproduction

TODO

### Other information

I suspect the `package.exports` might play a role here. SvelteKit doesn't use the TypeScript compiler option moduleResolution: Node16 or Bundler yet by default. It might be that the Vercel adapter ignores export maps altogether. A reproduction could be a mock package with exports fields and see how the Vercel adapter behaves.

**Comment by rishi-raj-jain** [comment]
@samuelstroschein 

Which repo is this regarding? I'm failing to see any payment-delivery file in the monorepo/

**Comment by samuelstroschein** [comment]
@rishi-raj-jain I asked Algora what a payment-delivery file is

**Comment by samuelstroschein** [comment]
@rishi-raj-jain a repo from a user. the issue can be replicated with a sveltekit starter template that uses the vercel adapter and deploys to vercel.

The bounty was meant for a community user of ours who faces the issue. Reproducing the error without context is likely difficult for you. Don't know if it's worth your time.

**Comment by rishi-raj-jain** [comment]
@samuelstroschein 

I'd love to know the issue exactly, and then help solve it. Does that sound good?

Where's the template where I can try to reproduce the ame?

**Comment by samuelstroschein** [comment]
@rishi-raj-jain you should be able to replicate the issue with this example https://github.com/inlang/monorepo/tree/main/inlang/source-code/paraglide/paraglide-js-adapter-sveltekit/example if you change the sveltekit adapter to vercel

**Comment by rishi-raj-jain** [comment]
Sure thing, on it!

**Comment by rishi-raj-jain** [comment]
@samuelstroschein 

The issue is with how you're deploying the app. I've just deployed it working at https://example-rishi-raj-jain.vercel.app/en.

As you'd know it needs the monorepo to refer to the packages, or are you deploying that template in an isolated environment which is seperate from it's monorepo? By default, if you just cd to the repo and deploy it to vercel, vercel uploads the code to their platform in case of which, only the starter is sent to Vercel's Code Source and not your repo, hence it's not able to find (as below).
<img width="948" alt="Screenshot 2023-10-25 at 4 04 17 PM" src="https://github.com/inlang/monorepo/assets/46300090/aeb3f3ea-8d3a-4a2f-bc5d-7d9048ea2ba1">


If so, you should update the workflows to do something like `vercel deploy --prebuilt` after running `npm run _build` in the monorepo (after replaced the adapter that is).

<img width="1512" alt="Screenshot 2023-10-25 at 4 03 06 PM" src="https://github.com/inlang/monorepo/assets/46300090/93e854af-cacf-4eb5-a4a9-88889cc3f19b">
<img width="695" alt="Screenshot 2023-10-25 at 4 04 35 PM" src="https://github.com/inlang/monorepo/assets/46300090/ab9f8043-c864-437b-86b3-2e970db72c8d">

**Comment by samuelstroschein** [comment]
@rishi-raj-jain, interesting observation that you can deploy to vercel. 

I should have clarified that we don't have deployment issues. Users (with own repositories;not our monorepo) have deployment issues. See [this discord thread](https://discord.com/channels/897438559458430986/1083724234142011392/1164227706267910165). The fix therefore seems unrelated. If you really want to solve this issue, try to deploy an individual repository to vercel that fetches the paraglide dependency from NPM. 

Maybe the user in the discord thread can share the repo with you. From my observation, the root cause seems that the vercel adapter does not bundle paraglide-js as indicated by the [screenshot](https://github.com/inlang/monorepo/issues/1485#issue-1949157371). 

Sorry for the lack of context, the bounty was initially meant for the community member in the Discord thread.

**Comment by rishi-raj-jain** [comment]
@samuelstroschein 

You'd have to first send me an invite link to the discord?

Here's what I'm seeing:

<img width="1451" alt="Screenshot 2023-10-25 at 5 20 11 PM" src="https://github.com/inlang/monorepo/assets/46300090/b8d3ae69-bfaa-42d4-b2eb-e53aeae68e0c">

**Comment by rishi-raj-jain** [comment]
> Sorry for the lack of context, the bounty was initially meant for the community member in the Discord thread.

Would you be open to giving it to me if I solve it?

**Comment by doic** [comment]
I hope @samuelstroschein can give you the bounty @rishi-raj-jain :-)
It was created for me, but if you can solve it, I'm totally happy with that ;-)
I'll give you access to my repo if you want to clone it and deploy it on Vercel.

**Comment by samuelstroschein** [comment]
> Would you be open to giving it to me if I solve it? 

Yes, of course!

**Comment by rishi-raj-jain** [comment]
Awesome, on it.

**Comment by rishi-raj-jain** [comment]
Please assign this to me?

**Comment by rishi-raj-jain** [comment]
This is solved!

- Deployed: https://uu-rishi-raj-jain.vercel.app/en
- Code (attached)

Steps to run:
- npm install
- npm run _build
- vercel deploy
 
[sveltekit-example.zip](https://github.com/inlang/monorepo/files/13169752/sveltekit-example.zip)

**Comment by rishi-raj-jain** [comment]
I don't think PRing the git repo was part of the bounty as you said it works for you, but I tried to set that repo locally and that too failed, so made fixes to that as well.

**Comment by rishi-raj-jain** [comment]
# TLDR

The main change is
```
ssr: {
     noExternal: ['@inlang/paraglide-js']
}
```

# Explanation in detail
As the compiled-output folder is already transpiled, using [noExternal with vite](https://vitejs.dev/guide/ssr.html#ssr-externals:~:text=If%20a%20dependency%20needs%20to%20be%20transformed%20by%20Vite%27s%20pipeline%2C%20for%20example%2C%20because%20Vite%20features%20are%20used%20untranspiled%20in%20them%2C%20they%20can%20be%20added%20to%20ssr.noExternal.) becomes necessary to be used so that is not left out while bundling.

More specifically, the issue was due to exclusion of the file `messages.js` in the node_modules of the vercel output while treeshaking, while `runtime.js` was included because as being the default export from the package.
